# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1262452141 18000
#      Sat Jan 02 12:09:01 2010 -0500
# Node ID e6441e13acb28accb0da2d907cf70c52445c617a
# Parent  60904110acbb749c30aa07e9148b4c2eff0268d7
lex.ll: Mac OSX's flex converts EOF to 0.

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2010-01-02  Ben Abbott <bpabbott@mac.com>
+
+	* lex.ll: Mac OSX's flex converts EOF to 0.
+
 2009-12-30  Jaroslav Hajek  <highegg@gmail.com>
 
 	* DLD-FUNCTIONS/cellfun.cc (scalar_query_helper): Remove.
 	(can_extract): New overloaded helper function.
 	(scalar_col_helper_nda): Rewrite using can_extract and
 	octave_value_extract. Instantiate for integer types.
 	(make_col_helper): Support integer types.
 
diff --git a/src/lex.ll b/src/lex.ll
--- a/src/lex.ll
+++ b/src/lex.ll
@@ -1031,17 +1031,21 @@ NUMBER	(({D}+\.?{D}*{EXPON}?)|(\.{D}+{EX
 
 . {
     LEXER_DEBUG (".");
 
     xunput (yytext[0], yytext);
 
     int c = text_yyinput ();
 
+#if defined (__APPLE__) && defined (__MACH__)
+    if (! (c == EOF || c == 0))
+#else /* not MacOS X */
     if (c != EOF)
+#endif
       {
 	current_input_column++;
 
 	error ("invalid character `%s' (ASCII %d) near line %d, column %d",
 	       undo_string_escape (static_cast<char> (c)), c,
 	       input_line_number, current_input_column);
 
 	return LEXICAL_ERROR;
